app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,61 @@
+class PasswordResetsController < ApplicationController
+  before_action :provide_user, only: [:edit, :update]
+  before_action :valid_user?, only: [:edit, :update]
+  before_action :check_expiration, only: [:edit, :update]
+
+  def new; end
+
+  def create
+    @user = User.find_by(email: params[:password_reset][:email].downcase)
+    if @user
+      @user.create_reset_digest!
+      @user.send_password_reset_email
+      flash[:info] = 'Check your email for password instructions'
+
+      redirect_to root_path
+    else
+      flash[:info] = 'Email address not found.'
+
+      redirect_to new_password_reset_path
+    end
+  end
+
+  def edit; end
+
+  def update
+    if params[:user][:password].empty?
+      @user.errors.add(:password, "can't be empty")
+      render 'edit'
+    elsif @user.update(user_params)
+      reset_session
+      @user.update_attribute(:reset_digest, nil)
+      log_in(@user)
+      flash[:success] = 'Password has been reset successfully!'
+
+      redirect_to @user
+    else
+      render 'edit'
+    end
```

Comment:
could you maybe split up the code into different methods? It is hard to follow when there are multiple if elseif else branches